internal/grpc/services/storageprovider/storageprovider.go

```diff
@@ -291,14 +291,38 @@ func (s *service) InitiateFileUpload(ctx context.Context, req *provider.Initiate
 		}, nil
 	}
 
+	metadata := map[string]string{}
+	ifMatch := req.GetIfMatch()
+	if ifMatch != "" {
+		sRes, err := s.Stat(ctx, &provider.StatRequest{Ref: req.Ref})
+		if err != nil {
+			return nil, err
+		}
+
+		switch sRes.Status.Code {
+		case rpc.Code_CODE_OK:
+			if sRes.Info.Etag != req.GetIfMatch() {
```

Comment:
Any reason why we are calling `req.GetIfMatch()` again instead of using `ifMatch` variable?

Reply:
good catch! :+1: